lib/pure/random.nim

```diff
@@ -635,20 +653,60 @@ when not defined(nimscript) and not defined(standalone):
       let time = int64(times.epochTime() * 1000) and 0x7fff_ffff
       result = initRand(time)
     else:
-      let now = times.getTime()
-      result = initRand(convert(Seconds, Nanoseconds, now.toUnix) + now.nanosecond)
+      proc getRandomState(): Rand =
```

Comment:
move this nested proc to top-level scope, simpler/cleaner

Reply:
Why moving nested proc to top-level scope makes simpler/cleaner?
It doesn't reduce a total number of lines of random module.
`getRandomState` proc is used only in `initRand()`.